src/main/java/cloud/eppo/EppoHttpClient.java

```diff
@@ -83,13 +83,11 @@ public void onResponse(Call call, Response response) {
                     callback.onFailure("Failed to read response from URL " + httpUrl);
                   }
                 } else {
-                  switch (response.code()) {
-                    case HttpURLConnection.HTTP_FORBIDDEN:
-                      callback.onFailure("Invalid API key");
-                      break;
-                    default:
-                      log.debug("Fetch failed with status code: " + response.code());
-                      callback.onFailure("Bad response from URL " + httpUrl);
+                  if (response.code() == HttpURLConnection.HTTP_FORBIDDEN) {
```

Comment:
According to IntelliJ (and I agree!) A `switch` with a single case should be an `if` 🤓

Reply:
😎